pkg/kubelet/config/http.go

```diff
@@ -183,6 +229,63 @@ func tryDecodeList(data []byte) (parsed bool, manifests []v1beta1.ContainerManif
 	return true, manifests, pods, nil
 }
 
+func tryDecodeSinglePod(data []byte) (parsed bool, pod api.Pod, err error) {
+	// TODO(wojtek-t): Should we support pods in different api versions?
```

Comment:
You're looking for "api.Scheme.Decode()" from pkg/api. A good example is "cmd/kube-version-change/version.go" which converts from any version to any other version.